suggested edits (#68)
poseidon-framework/poseidon-framework.github.io: janno_details.md

The `Date_Type` column handles the general distinction between the most common forms of age information:

- `modern`: Applies to present-day reference samples, so not ancient DNA.
- `C14`: Applies if there is a set of radiocarbon dates explicitly listed in the columns `Date_C14_Labnr`, `Date_C14_Uncal_BP` and `Date_C14_Uncal_BP_Err` whose post-calibration probability distribution is a meaningful prior for the individual’s year of death. The dates do not always have to be directly from the individual's tissue, but they should be immediately relevant for their year of death (e.g. a date from a grain kernel recovered from the individual's grave).
- `contextual`: Applies in all other cases if the columns `Date_BC_AD_Median`, `Date_BC_AD_Start`, `Date_BC_AD_Stop` can be filled. This includes age attribution based on the archaeologically determined stratigraphy or typological information. `contextual` should also be chosen if the sample is dated very indirectly with radiocarbon dating (e.g. radiocarbon dates from other, unrelated features of the same site) or dated with other physical or chemical dating methods (e.g. dendrochronology or optically stimulated luminescence).

> So `Date_C14_Labnr`, `Date_C14_Uncal_BP` and `Date_C14_Uncal_BP_Err` only go along with `Date_Type = C14`, whereas `Date_BC_AD_Median`, `Date_BC_AD_Start`, `Date_BC_AD_Stop` complement both `Date_Type = C14` and `Date_Type = contextual`. Radiocarbon dates that only serve as secondary evidence for a contextual dating should NOT be reported in `Date_C14_Labnr`, `Date_C14_Uncal_BP` and `Date_C14_Uncal_BP_Err`.

Author: if we want to stick to that nomenclature in this doc, we need to add a link to the reference, and capitalise all the definition keywords.

The column `Date_Note` stores arbitrary free-form text information about the dating of a sample.

### Genetic summary data

#### Individual properties

The `Genetic_Sex` column should encode the biological sex as determined from the DNA read distribution on the X and Y chromosome. It only allows for the entries

- `F`: female
- `M`: male
- `U`: unknown

This limitation stems from the genotype data formats by Plink and the Eigensoft software package. Edge cases (e.g. XXY, XYY, X0, ...) can not be expressed with this format and should be reported as `U` with an additional comment in the free text `Note` field. Genetic sex determination for ancient DNA can be performed for example with Sex.DetERRmine [@Lamnidis2018](https://doi.org/10.1038/s41467-018-07483-5).

The `MT_Haplogroup` column is meant to store the human mitochondrial DNA haplogroup for the respective individual in a simple string. The entry can be arbitrarily precise. A software tool to determine the MT haplogroup is for example Haplogrep [@Schnoeherr2023](https://doi.org/10.1093/nar/gkad284).

> The `Y_Haplogroup` column holds the respective human Y-chromosome DNA haplogroup in a simple string. The notation should follow a syntax with the main branch + the most terminal derived Y-SNP separated with a minus symbol (e.g. R1b-P312), following XXXX naming system.

Author: ```suggestion
The `Y_Haplogroup` column holds the respective human Y-chromosome DNA haplogroup in a simple string. To avoid confusion from using different haplotype naming systems, the notation should follow a syntax with the main branch + the most terminal derived Y-SNP separated with a minus symbol (e.g. R1b-P312), similar to that used by [Yfull](https://www.yfull.com/sc/tree/).
```

The `UDG` column documents if the libraries for the respective individual went through UDG (or USER enzyme) treatment. This wet lab protocol step removes molecular damage in the form of deaminated cytosines characteristic of ancient DNA.

- `minus`: A protocol without UDG treatment (e.g. [@Aron2019](https://doi.org/10.17504/protocols.io.bakricv6)).
- `half`: A protocol with UDG-half treatment (e.g. [@Aron2020a](https://doi.org/10.17504/protocols.io.bmh6k39e)).
- `plus`: A protocol with UDG-full treatment (e.g. [@Aron2020b](https://doi.org/10.17504/protocols.io.bqbpmsmn)).
- `mixed`: Multiple libraries that went through different UDG treatment approaches, and whose data were later merged. It is preferred that in such cases, this column is formatted as a list column, specifying the udg treatment of each individual library, in the order used in `Library_Names`.

The `Library_Built` column describes the library preparation method regarding single- or double-stranded protocols. See e.g. [@Gansauge2013](https://doi.org/10.1038/nprot.2013.038) for more information.

- `ds`: Double-stranded library preparation.
- `ss`: Single-stranded library preparation.

> - `mixed`: If multiple libraries with different strandedness were combined. It is preferred that in such cases, this column is formatted as a list column, specifying the strandedness of each individual library, in the order used in `Library_Names`. See also the Sequencing Source File in the Poseidon package as a way to provide details.

Author: Sorry. I seem to have mixed something up. I remember us discussing removing the `mixed` option, though I realise now that was for the SSF schema. I will remove this.

- `father_of`: This individual is likely the father of the partner individual
- `grandchild_of`: This individual is likely the grandchild of the partner individual
- `mother_or_daughter_of`: This individual is likely either the mother or daughter of the partner individual (which might be unclear, in case of imprecise archaeological dating)
- `unknown`: The relationship is unclear or not yet determined. This is the default state and does not have to be expressed, unless multiple relationships are present and some but not all are known.
- `...`

> Unlike `Relation_Degree`, `Relation_Type` can be left empty even if there are entries in `Relation_To`. But if it is filled, then the number of values must be equal to the number of entries in both `Relation_To` and `Relation_Degree`.

Author: Since `Relation_Type` HAS to be a list of length `n(Relation_Degree` if there are multiple relationships, that implicitly means unknown relationships will also have to be explicitly stated in some cases. 

It would be good to offer the suggested way to do so (see my suggestion for L40). 

To clarify, if we know one relationship is `father_of` but are unsure about the other 2 we have identified, would we prefer:
 - `father_of;;` or
 - `father_of;unknown;unknown`

I went for the latter in my suggestion

Reviewer: Sounds good to me!

- `half`: A protocol with UDG-half treatment (e.g. [@Aron2020a](https://doi.org/10.17504/protocols.io.bmh6k39e)).
- `plus`: A protocol with UDG-full treatment (e.g. [@Aron2020b](https://doi.org/10.17504/protocols.io.bqbpmsmn)).
- `mixed`: Multiple libraries that went through different UDG treatment approaches, and whose data were later merged. It is preferred that in such cases, this column is formatted as a list column, specifying the udg treatment of each individual library, in the order used in `Library_Names`.

The `Library_Built` column describes the library preparation method regarding single- or double-stranded protocols. See e.g. [@Gansauge2013](https://doi.org/10.1038/nprot.2013.038) for more information.

- `ds`: Double-stranded library preparation.
- `ss`: Single-stranded library preparation.
- `mixed`: If multiple libraries with different strandedness were combined. It is preferred that in such cases, this column is formatted as a list column, specifying the strandedness of each individual library, in the order used in `Library_Names`. See also the Sequencing Source File in the Poseidon package as a way to provide details.

> The `Genotype_Ploidy` column stores a characteristic of the aDNA genotyping procedure. Humans have two complete sets of chromosomes in their cells and hence are diploid organisms. For many computational aDNA applications it is more practical, though, to work with "pseudo-haploid" data. This is a procedure, where a haploid genotype call is produced (often through a random-draw procedure) and then "doubled" to create the diploid genotype of the individual. Confusingly, genotypes produced in this way are also referred to as "pseudo-diploid". The `Genotype_Ploidy` column can contain one of the following values:

Author: LGTM

- `diploid`: True diploid genotype calls were made.
- `haploid`: Haploid genotypes were called and then doubled.

The column `Data_Preparation_Pipeline_URL` should finally store an URL that links to a complete and human-readable description of the computational pipeline (for example a specific configuration for nf-core/eager [@FellowsYates2021](https://doi.org/10.7717/peerj.10947)) by which the sample data was processed.

#### Data yield

The `Endogenous` column holds the percentage of mapped reads over the total amount of reads that went into the mapping pipeline. That boils down to the DNA percentage of the library that matches the (human) reference. It should be determined from Shotgun libraries (so before any hybridization capture), not on target (i.e. across the whole genome, not specific positions), and before any mapping quality filtering. In case of multiple libraries only the highest value should be reported. The % endogenous DNA can be calculated for example with the [endorS.py](https://github.com/aidaanva/endorS.py) script.

The `Nr_SNPs` column gives the number of SNPs reported in the genotype data files for this individual.

> The `Coverage_on_Target_SNPs` column reports the mean fold coverage on the SNP set of the genotype dataset (e.g. 1240K) for the merged libraries of this sample. To calculate the coverage it is necessary to determine which SNPs are covered how many times by the mapped reads. Individual SNPs might be covered multiple times, whereas others may not be covered at all by the highly deteriorated ancient DNA. The coverage for each SNP is therefore a number between 0 and n. The statistic can be determined for example with the QualiMap [@Okonechnikov2015](https://doi.org/10.1093/bioinformatics/btv566) software package. In case of multiple libraries, the coverage can be given as a mean across all of them.

Author: This is not right. When compiling multiple coverages, the reported value should not be the mean, but probably the sum?
If I have a library with 2X 1240k coverage and another with 1X, merging them would give an average coverage of 3X, not 1.5X. 
[janno_columns.tsv](https://github.com/poseidon-framework/poseidon-schema/blob/master/janno_columns.tsv) might also need updating accordingly.

Reviewer: The definition in the schema says

> average X-fold coverage across targeted SNP sites after quality filtering

I think this does not say anything about what should be done in case of multiple libraries, so we're safe there, as far as my understanding goes?

Now regarding the statement here:

> In case of multiple libraries, the coverage can be given as a mean across all of them.

I guess you're right. The mean doesn't make much sense here. It's rather the maximum or something between the maximum and the sum, right? I think we can just omit this sentence entirely.

Reviewer 2: Well, I think multiple libraries are typically merged (on a stream or explicitly) when piped into a caller, so I agree that I think it should be the sum. This is somewhat awkward, though, if different subsets of libraries are subjected to different calling algorithms (as for example in-house with single- and double-stranded data). But even then, I think the sum of the coverages is somewhat what tells us something about the amount of data to expect. So yes, I think it should be the sum. I suggest:

```suggestion
The `Coverage_on_Target_SNPs` column reports the mean fold coverage on the SNP set of the genotype dataset (e.g. 1240K) for the merged libraries of this sample. To calculate the coverage it is necessary to determine which SNPs are covered how many times by the mapped reads. Individual SNPs might be covered multiple times, whereas others may not be covered at all by the highly deteriorated ancient DNA. The coverage for each SNP is therefore a number between 0 and n. The statistic can be determined for example with the QualiMap [@Okonechnikov2015](https://doi.org/10.1093/bioinformatics/btv566) software package. In case of multiple libraries, the total coverage should be given across all libraries.
```

Reviewer 3: can we change 1240K to 1240k because reasons?

Author: 1240K is the code that has been defined for this snp set across Poseidon, including in the POSEIDON.yml. I think consistency is more important than technical correctness. We're in too deep by now.

Reviewer 2: Oh boy, so sorry, @wolfgangaroo. I think that ship has sunken already.

The column `Date_Note` stores arbitrary free-text information about the dating of a sample.

### Genetic summary data

#### Individual properties

The `Genetic_Sex` column should encode the biological sex as determined from the DNA read distribution on the X and Y chromosome. It only allows for the entries

- `F`: female
- `M`: male
- `U`: unknown

This limitation stems from the genotype data formats by Plink and the Eigensoft software package. Edge cases (e.g. XXY, XYY, X0, ...) can not be expressed with this format and should be reported as `U` with an additional comment in the free text `Note` field. Genetic sex determination for ancient DNA can be performed for example with Sex.DetERRmine [@Lamnidis2018](https://doi.org/10.1038/s41467-018-07483-5).

The `MT_Haplogroup` column is meant to store the human mitochondrial DNA haplogroup for the respective individual in a simple string. The entry can be arbitrarily precise. A software tool to determine the MT haplogroup is for example Haplogrep [@Schnoeherr2023](https://doi.org/10.1093/nar/gkad284).

> The `Y_Haplogroup` column holds the respective human Y-chromosome DNA haplogroup in a simple string. The notation should follow a syntax with the main branch + the most terminal derived Y-SNP separated with a minus symbol (e.g. R1b-P312), following XXXX naming system.

Author: The question is how the _main branch_ itself is named. I believe that depending on the reference dataset used, the branches can have slightly different names, meaning that the same sequence could be `N3a-SNP2` in one nomenclature while being `N1c1-SNP2` in the other.

> The `Y_Haplogroup` column holds the respective human Y-chromosome DNA haplogroup in a simple string. The notation should follow a syntax with the main branch + the most terminal derived Y-SNP separated with a minus symbol (e.g. R1b-P312), following XXXX naming system.

Reviewer 3: It's neither N1c1 nor N3a, but rather N1a, but it's more complicated. ISOGG hasn't been updated since 2020 and the alpha-numerical system has proven impractical for a haplotypic system the size of even the tiny Y chromosome.
Yfull resides to just using the main letter N (for rough orientation on the tree) follow by the terminal SNP.
I feel we have no other choice than following a similar approach as the Y-SNPs are the key identifier.

> The `Y_Haplogroup` column holds the respective human Y-chromosome DNA haplogroup in a simple string. The notation should follow a syntax with the main branch + the most terminal derived Y-SNP separated with a minus symbol (e.g. R1b-P312), following XXXX naming system.

Author: Then maybe suggeting using the Y-tree nomenclature is already enough, potentially also providing an example. I don't think it is realistic to recode all existing Y haplogroups into that format, or validate the nomenclature in any way though.

The `Primary_Contact` column is a free-form text field that stores the name of the main or the corresponding author of the respective paper for published data.

The `Publication` column holds either the value `unpublished` for (yet) unpublished samples or -- for published data -- one or multiple citation-keys of the form `AuthorJournalYear` without any spaces or special characters. These keys have to be identical to the [BibTeX](http://www.bibtex.org) citation-keys identifying the respective entries in the `.bib` file of the package. BibTeX is a file format to store bibliographic information, where each entry (article, book, website, ...) is defined by a series of parameters (authors, year of publication, journal, ...). Here's an example `.bib` file with two entries for [@Cassidy2015](https://doi.org/10.1073/pnas.1518445113) and [@Feldman2019](https://doi.org/10.1126/sciadv.aax0061):

```default

> @article{CassidyPNAS2015,

Author: Updated these keys to match the text.

Reviewer: OK!

Reviewer: Wait. The text has `@Cassidy2015` and `@Feldman2019` above and `CassidyPNAS2015;FeldmanScienceAdvances2019` below. Maybe we rather adjust that.

Author: I believe @Cassidy2015 and @Feldman2019 refer to the bibtex of the manuscript, while `CassidyPNAS2015;FeldmanScienceAdvances2019` to the bibtex we're showcasing (this is clearer in the PDF version).
The poseidon schema is clear that `CassidyPNAS2015;FeldmanScienceAdvances2019` is the right formatting for Poseidon (`AuthorJournalYear`), so i believe editing the shown bibtex is the right way to go.

Reviewer: OK - sounds good to me :+1:
